tests/integration/incremental_unique_id_test/test_incremental_unique_id.py

```diff
@@ -0,0 +1,270 @@
+from dbt.exceptions import ParsingException
+from tests.integration.base import DBTIntegrationTest, use_profile
+from collections import namedtuple
+from pathlib import Path
+
+
+TestResults = namedtuple(
+    'TestResults',
+    ['seed_count', 'model_count', 'seed_rows', 'inc_test_model_count',
+     'opt_model_count', 'relation'],
+)
+
+
+class TestIncrementalUniqueKey(DBTIntegrationTest):
+    @property
+    def schema(self):
+        return 'incremental_unique_key'
+
+    @property
+    def models(self):
+        return 'models'
+
+    def count_test_relations(self, seed, incremental_model):
+        '''Idempotently create some number of seeds and incremental models'''
+        seed_result_set = self.run_dbt(
+            ['seed', '--select', seed, '--full-refresh']
+        )
+        model_result_set = self.run_dbt(
+            ['run', '--select', incremental_model, '--full-refresh']
+        )
+        return len(seed_result_set), len(model_result_set)
+
+    def get_seed_rows_after_update(self, seed, update_sql_file):
+        '''Upate seed and return new row count'''
+        row_count_query = 'select * from {}.{}'.format(
+            self.unique_schema(),
+            seed
+        )
+
+        self.run_sql_file(Path('seeds') / Path(update_sql_file + '.sql'))
+        return len(self.run_sql(row_count_query, fetch='all'))
+
+    def update_incremental_model(self, incremental_model):
+        '''update incremental model after the seed table has been updated'''
+        model_result_set = self.run_dbt(['run', '--select', incremental_model])
+        return len(model_result_set)
+
+    def setup_test(self, seed, incremental_model, update_sql_file):
+        '''build a test case and return values for assertions'''
+        (seed_count, model_count) = self.count_test_relations(
+            seed=seed, incremental_model=incremental_model
+        )
+        seed_rows = self.get_seed_rows_after_update(
+            seed=seed, update_sql_file=update_sql_file
+        )
+        inc_test_model_count = self.update_incremental_model(
+            incremental_model=incremental_model
+        )
+
+        return (seed_count, model_count, seed_rows, inc_test_model_count)
+
+    def run_test(self, expected_fields, test_case_fields):
```

Comment:
This part doesn't run things but more like comparing result? Should we consider updating the name a bit?

Reply:
That's a fair callout! Hmmm, maybe something like test_scenario_correctness?

Reply:
Sounds good to me. Naming is hard